datasets/tf_datasets.py

```diff
@@ -0,0 +1,117 @@
+"""
+https://www.tensorflow.org/datasets
+"""
+
+
+from typing import Optional
+from sisyphus import *
+
+
+class DownloadAndPrepareTfDatasetJob(Job):
+    """
+    https://www.tensorflow.org/datasets
+
+        pip install tensorflow-datasets
+
+    It further needs some extra dependencies, for example for 'librispeech':
+
+        pip install apache_beam
+        pip install pydub
+        # ffmpeg installed
+
+    See here for some more:
+    https://github.com/tensorflow/datasets/blob/master/setup.py
+
+    Also maybe::
+
+        pip install datasets  # for Huggingface community datasets
+    """
+
+    def __init__(
+        self,
+        dataset_name: str,
+        *,
+        max_simultaneous_downloads: Optional[int] = None,
+        max_workers: Optional[int] = None,
+    ):
+        """
+        :param dataset_name:
+            https://www.tensorflow.org/datasets/overview
+            https://www.tensorflow.org/datasets/catalog/overview
+            https://www.tensorflow.org/datasets/community_catalog/huggingface
+        :param max_simultaneous_downloads: https://github.com/tensorflow/datasets/issues/3885
+        :param max_workers:
+            For download extractor (https://github.com/tensorflow/datasets/issues/3887)
+            and Apache Beam.
+        """
+        super().__init__()
+        self.dataset_name = dataset_name
+        self.max_simultaneous_downloads = max_simultaneous_downloads
+        self.max_workers = max_workers
+
+        self.out_data_dir = self.output_path("data_dir")
+
+    @classmethod
+    def hash(cls, kwargs):
+        # All other options except dataset_name are ignored, as they should not have an influence on the result.
+        d = {
+            "dataset_name": kwargs["dataset_name"],
+        }
+        return super().hash(d)
+
+    def tasks(self):
+        yield Task("run", resume="run", mini_task=True)
```

Comment:
The `resume` parameter does not make sense for mini-tasks, as it is only called when the job did not crash but goes "missing" from the SGE. The `localengine` which is used in case of `mini_task=True` has no support for `gs.STATE_INTERRUPTED`, which is the only case in which resume can be used.